### PR TITLE
[DO NOT MERGE] Tmroot

### PR DESCRIPTION
**Pull request by benjaminbollen**
- [ ] 1. bring e-cm and e-db in line
- [x] 2. move tendermint to the status of a consensus module
- [x] 3. clean up config and command line interface for eris-db with cobra and viper
- [ ] 4. clean up DOCKER and shell scripts interface into eris-db

out-of-scope extras
a. import event subscription infrastructure from go-ethereum

### CODE REVIEW

File: DOCKER/start.sh [hunk 1]
```diff
@@ -41,7 +41,7 @@ fi
 CHAIN_DIR="/home/$USER/.eris/chains/$CHAIN_ID"
 
 # set the tendermint directory
-TMROOT=$CHAIN_DIR
+TMROOT="$CHAIN_DIR"/tendermint
```
Comment (benjaminbollen) [inline DOCKER/start.sh]: typo noted

File: event/event.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+)
+
+// Event is a time-tagged notification pushed to subscribers.
+type Event struct {
```
Comment (silasdavis) [inline event/event.go]: So we're copying this in, not depending on Ethereum for it? I can see a rationale as it changes for different reasons and it's a fairly clean piece of code, but worth considering the alternative.

Comment (benjaminbollen) [inline event/event.go]: True, worthwhile consideration but i am of the opinion that here we are better served taking ownership of this code:
1. As you mentioned it is a clean, no extra dependencies, clean piece of code - thank you geth
2. The event switch is absolutely crucial for an event based architecture; so i see many dangers considering the event switch third party code

Comment (silasdavis) [inline event/event.go]: OK, I'm sold.

File: definitions/do.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+func NowDo() *Do {
+	do := new(Do)
+	do.Debug = false
+	do.Verbose = false
+	// the default value for output is set to true in cmd/eris-db.go;
+	// avoid double setting it here though
+	do.Output = false
+  do.ChainId = ""
```
Comment (silasdavis) [inline definitions/do.go]: indent

File: definitions/do.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  viper "github.com/spf13/viper"
+)
+
+type Do struct {
+  // Persistent flags not reflected in the configuration files
+  // only set through command line flags or environment variables
+	Debug        bool     // ERIS_DB_DEBUG
+	Verbose      bool     // ERIS_DB_VERBOSE
+	Output       bool     // ERIS_DB_OUTPUT
+  // Capital configuration options explicitly extracted from the Viper config
```
Comment (silasdavis) [inline definitions/do.go]: You mean they appear here an in config? How do we protect from divergence?

File: cmd/eris-db.go [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+)
+
+const VERSION = version.VERSION
+
+// Global Do struct
+var do *definitions.Do
+
+var ErisDbCmd = &cobra.Command {
+  Use:   "eris-db",
+  Short: "Eris-DB is the beating heart of the eris chain.",
```
Comment (silasdavis) [inline cmd/eris-db.go]: Emosh

File: definitions/do.go (hunk 4 above)
Comment (benjaminbollen) [inline definitions/do.go]: no debug, verbose and output are not set in config; flags that are reflected in config will have to be covered by viper

File: cmd/eris-db.go (hunk 5 above)
Comment (benjaminbollen) [inline cmd/eris-db.go]: made with <3

**Comment by silasdavis** [comment]
so this LGTM, but assuming WIP means you don't want it merged yet

**Comment by benjaminbollen** [comment]
Closing this PR as it is not ready to move into `develop-tmsp`.  These commits are pushed at branch `discuss-eris-gateway` as the essentials of ErisMint as an application are tied up with Tendermint as an in-process consensus engine.
At this point the E-DB servers need to be loaded in this new structure and the RPC calls need to be separated to application and consensus behind the pipe.

**Pull request closed without merging**